src/backend/distributed/transaction/remote_transaction.c

```diff
@@ -388,6 +392,13 @@ StartRemoteTransactionPrepare(struct MultiConnection *connection)
 
 	Assign2PCIdentifier(connection);
 
+	/* log transactions to workers in pg_dist_transaction */
+	workerNode = FindWorkerNode(connection->hostname, connection->port);
+	if (workerNode != NULL)
+	{
+		LogTransactionRecord(workerNode->groupId, transaction->preparedName);
```

Comment:
I think it's great that we use this more generally now, but I do think we should have some tests for recovery for !MX stuff.